core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -2064,33 +2108,135 @@ private static void publishOpenColumnTasks(
                     timestampMergeIndexSize = dedupRows * TIMESTAMP_MERGE_ENTRY_BYTES;
                     timestampMergeIndexAddr = Unsafe.realloc(tempIndexAddr, tempIndexSize, timestampMergeIndexSize, MemoryTag.NATIVE_O3);
                     final long duplicateCount = mergeRowCount - dedupRows;
+                    boolean appendOnly = false;
                     if (duplicateCount > 0) {
-                        // we could be de-duping a split partition
-                        // in which case only its size will be affected
+                        if (duplicateCount == mergeOOOHi - mergeOOOLo + 1 && prefixType != O3_BLOCK_O3) {
+
+                            // All the rows are duplicates, the commit does not add any new lines.
+                            // Check non-key columns if they are exactly the same as the rows they replace
+                            if (tableWriter.checkDedupCommitIdenticalToPartition(
+                                    oldPartitionTimestamp,
+                                    srcNameTxn,
+                                    srcDataOldPartitionSize,
+                                    mergeDataLo,
+                                    mergeDataHi,
+                                    mergeOOOLo,
+                                    mergeOOOHi,
+                                    timestampMergeIndexAddr,
+                                    dedupRows
+                            )) {
+
+                                if (suffixType != O3_BLOCK_O3) {
+                                    LOG.info().$("deduplication resulted in noop [table=").$(tableWriter.getTableToken())
+                                            .$(", partition=").$ts(partitionTimestamp)
+                                            .I$();
+
+                                    // nothing to do, skip the partition
+                                    updatePartition(
+                                            tableWriter.getFilesFacade(),
+                                            srcTimestampAddr,
+                                            srcTimestampSize,
+                                            srcTimestampFd,
+                                            tableWriter,
+                                            partitionUpdateSinkAddr,
+                                            oldPartitionTimestamp,
+                                            Long.MAX_VALUE,
+                                            -1,
+                                            srcDataOldPartitionSize,
+                                            0
+                                    );
+                                    Unsafe.free(timestampMergeIndexAddr, timestampMergeIndexSize, MemoryTag.NATIVE_O3);
```

Comment:
Zeroing a dangling address is always a good idea. Same applies to L2202 and L2250
```suggestion
                                    timestampMergeIndexAddr = Unsafe.free(timestampMergeIndexAddr, timestampMergeIndexSize, MemoryTag.NATIVE_O3);
```

Reply:
Done in https://github.com/questdb/questdb/pull/5764/commits/0b1f3303942d1609a5563f612cc40f32b8901c3d, except L2250. There, it is highlighted as a redundant assignment.